### PR TITLE
Added gt/ltrharvest

### LINKED ISSUE #4527: new module: gt/ltrharvest
### Is there an existing module for this?

- [X] I have searched for the existing module

### Is there an open PR for this?

- [X] I have searched for existing PRs

### Is there an open issue for this?

- [X] I have searched for existing issues

### Are you going to work on this?

- [X] If I'm planning to work on this module, I added myself to the `Assignees` to facilitate tracking who is working on the module

### PR DESCRIPTION
**Pull request by GallVp**
## PR checklist

Closes #4527  

- [x] Added gt/ltrharvest
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/gt/ltrharvest/main.nf [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        'biocontainers/genometools-genometools:1.6.5--py310h3db02ab_0' }"
+
+    input:
+    tuple val(meta), path(index)
+
+    output:
+    tuple val(meta), path("*.tabout")       , emit: tabout      , optional: true
+    tuple val(meta), path("*.gff3")         , emit: gff3        , optional: true
+    tuple val(meta), path("out.fasta")      , emit: fasta       , optional: true    // When args has -out out.fasta
+    tuple val(meta), path("inner.fasta")    , emit: inner_fasta , optional: true    // When args has -outinner inner.fasta
```
Comment (mahesh-panchal) [inline modules/nf-core/gt/ltrharvest/main.nf:17]: Are these the default names, or can they be named using `ext.args`? If they can be named using `ext.args`, then this will not work.

Comment (GallVp) [reply modules/nf-core/gt/ltrharvest/main.nf]: Thank you @mahesh-panchal 

Really good point. I had made the assumption that the user will always specify the names as out.fasta and inner.fasta in `ext.args`. Now I have removed that assumption and added multiple tests to verify the generalised code. There might be some edge case here that I am missing.

**Review by mahesh-panchal — APPROVED** [review]
